doc/discretization.rst

```diff
@@ -0,0 +1,312 @@
+&- \sum_{E\in\Omega_h}\left\lbrack\oint_{\partial E}\phi_j^k \b{h}_e(\b{Q}_h^+,
+\b{Q}^-_h; \b{n}) d\sigma + \oint_{\partial E} \phi_j^k \b{h}_v(\b{Q}_h^+,
+\b{\Sigma}_h^+, \b{Q}_h^-, \b{\Sigma}_h^-; \b{n}) d\sigma\right\rbrack \\
+&+ \sum_{E\in\Omega_h} \int_E \phi_j^k\b{S}_h d\Omega, \\
+\sum_{E\in\Omega_h}\int_E\phi_j^k \b{\Sigma}_h d\Omega &= \sum_{E\in\Omega_h}\left\lbrack
+\oint_{\partial{E}}\phi_j^k \b{H}_s(\b{Q}^+_h, \b{Q}_h^-; \b{n}) d\sigma -
+\int_E\nabla\phi^k_j\cdot\b{Q}_h d\Omega\right\rbrack.
+$$
+
+Numerical fluxes
+================
+
+To account for the discontinuities at element faces, DG employs numerical fluxes, which are
+enforced to be singled-valued, but are functions of both $\pm$ states:
+
+$$
+\b{h}_e(\b{Q}_h^+, \b{Q}^-_h; \b{n}) &\approx \b{F}^I(\b{Q}_h)
+\cdot\b{n}, \\
+\b{h}_v(\b{Q}_h^+, \b{\Sigma}_h^+, \b{Q}_h^-, \b{\Sigma}_h^-;
+\b{n}) &\approx \b{F}^V(\b{Q}_h, \b{\Sigma}_h)\cdot\b{n}\\
+\b{H}_s(\b{Q}^+_h, \b{Q}_h^-; \b{n}) &\approx \b{Q}_h\b{n}.
+$$
+
+Choices of numerical fluxes corresponding to BR1
+------------------------------------------------
+
+Numerical inviscid flux
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Typical choices for $\b{h}_e(\b{Q}_h^+, \b{Q}^-_h; \b{n})$ include,
+but are not limited to:
+
+* Local Lax-Friedrichs (LLF)
+* Roe
+* Engquist-Osher
+* HLLC
+
+|mirgecom| currently uses LLF, which is implemented as follows:
+$$
+\b{h}_{e}(\b{Q}_h^+, \b{Q}^-_h; \b{n}) = \frac{1}{2}\left(
+\b{F}^{I}(\b{Q}_h^+)+\b{F}^{I}(\b{Q}_h^-)\right) - \frac{\lambda}
+{2}\left(\b{Q}_h^+ - \b{Q}_h^-\right)\b{n},
+$$
+where $\lambda$ is the characteristic max wave-speed of the fluid. Numerical fluxes
+which penalize the ''jump'' of the state $\left(\b{Q}_h^+ - \b{Q}_h^-\right)
+\b{n}$ act as an additional source of dissipation, which has a stabilizing effect
+on the numerics.
+
+Numerical viscous flux
+^^^^^^^^^^^^^^^^^^^^^^
+Take $\b{h}_v(\b{Q}_h^+, \b{\Sigma}_h^+, \b{Q}_h^-, \b{\Sigma}_h^-; \b{n})$ to be
+defined in a ''centered'' (average) way:
+$$
+\b{h}_v(\b{Q}_h^+, \b{\Sigma}_h^+, \b{Q}_h^-, \b{\Sigma}_h^-;
+\b{n}) = \frac{1}{2}\left(\b{F}^V(\b{Q}_h^+, \b{\Sigma}_h^+) +
+\b{F}^V(\b{Q}_h^-, \b{\Sigma}_h^-)\right)\cdot\b{n}
+$$
+
+Numerical gradient flux
+^^^^^^^^^^^^^^^^^^^^^^^
+And similarly for the gradient flux:
+$$
+\b{H}_s(\b{Q}_h^+, \b{Q}_h^- ; \b{n}) = \frac{1}{2}\left(
+\b{Q}_h^+ + \b{Q}_h^-\right)\b{n}
+$$
+
+Domain boundary considerations
+------------------------------
+
+What happens when $\partial E \cap \partial\Omega \neq \emptyset$?
+
+In DG, numerical fluxes are not only responsible for handling the flow of information
+between adjacent cells, but they also enforce information flow at the boundaries.
+
+We denote the *boundary fluxes* as $\b{h}^*_e(\b{Q}_{bc})$,
+$\b{h}^*_v(\b{Q}_{bc}$, $\b{\Sigma}_{bc})$, and
+$\b{H}^*_s(\b{Q}_{bc})$, where $\b{Q}_{bc}$, $\b{\Sigma}_{bc}$ denote
+boundary conditions imposed on the state, and the gradient of the state respectively.
```

Comment:
Where doe these come from? What are they for solid walls?

Reply:
Is it sufficient to mention particular boundary treatments are described in the sections down below (0cc222c)?